diff --git a/dist/html-parse-stringify.d.ts b/dist/html-parse-stringify.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..771d92ac1d939ed00ce24b57b958a99912285a6a
--- /dev/null
+++ b/dist/html-parse-stringify.d.ts
@@ -0,0 +1,29 @@
+// Copied from https://github.com/HenrikJoreteg/html-parse-stringify/blob/master/html-parse-stringify.d.ts
+
+declare var htmlParseStringify: htmlParseStringify.htmlParseStringify;
+
+declare module htmlParseStringify {
+  export interface htmlParseStringify {
+    new (): htmlParseStringify;
+    parse_tag(tag: string): IDoc;
+    parse(html: string, options?: IOptions): IDoc[];
+    stringify(doc: IDoc): string;
+  }
+
+  export interface IDoc {
+    type: string;
+    content?: string;
+    voidElement: boolean;
+    name: string;
+    attrs: Record<string, unknown>;
+    children: IDoc[];
+  }
+
+  export interface IOptions {
+    components: string[];
+  }
+}
+
+declare module "html-parse-stringify" {
+  export = htmlParseStringify;
+}
diff --git a/package.json b/package.json
index 609d8e2dea6c7576d260f88749849337684757f1..752c717fd3a79af934d07b9c2cb5de975c3dd117 100644
--- a/package.json
+++ b/package.json
@@ -31,6 +31,7 @@
   "module": "dist/html-parse-stringify.module.js",
   "source": "src/index.js",
   "unpkg": "dist/html-parse-stringify.umd.js",
+  "types": "dist/html-parse-stringify.d.ts",
   "prettier": {
     "arrowParens": "avoid",
     "singleQuote": true,
