diff --git a/dist/html-parse-stringify.d.ts b/dist/html-parse-stringify.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..63fb1d0cba7b89e7cbe4a2f8037cefb6505fb4bd
--- /dev/null
+++ b/dist/html-parse-stringify.d.ts
@@ -0,0 +1,51 @@
+// Copied from https://github.com/HenrikJoreteg/html-parse-stringify/blob/master/html-parse-stringify.d.ts
+
+declare var htmlParseStringify: htmlParseStringify.htmlParseStringify;
+
+declare module htmlParseStringify {
+  export interface htmlParseStringify {
+    new (): htmlParseStringify;
+    parse_tag(tag: string): Node;
+    parse(html: string, options?: IOptions): Node[];
+    stringify(doc: Node): string;
+  }
+
+  export type Node = TextNode | TagNode | VoidTagNode | InterfaceNode;
+
+  interface TextNode {
+    type: "text";
+    content: string;
+  }
+
+  interface VoidTagNode {
+    type: "tag";
+    name: string;
+    attrs: Record<string, unknown>;
+    voidElement: true;
+    children: [];
+  }
+
+  interface TagNode {
+    type: "tag";
+    name: string;
+    attrs: Record<string, unknown>;
+    voidElement: false;
+    children: Node[];
+  }
+
+  interface ComponentNode {
+    type: "component";
+    name: string;
+    attrs: Record<string, unknown>;
+    voidElement: boolean;
+    children: [];
+  }
+
+  export interface IOptions {
+    components: string[];
+  }
+}
+
+declare module "html-parse-stringify" {
+  export = htmlParseStringify;
+}
diff --git a/package.json b/package.json
index 609d8e2dea6c7576d260f88749849337684757f1..752c717fd3a79af934d07b9c2cb5de975c3dd117 100644
--- a/package.json
+++ b/package.json
@@ -31,6 +31,7 @@
   "module": "dist/html-parse-stringify.module.js",
   "source": "src/index.js",
   "unpkg": "dist/html-parse-stringify.umd.js",
+  "types": "dist/html-parse-stringify.d.ts",
   "prettier": {
     "arrowParens": "avoid",
     "singleQuote": true,
